File: examples/mcp-client.mjs
#!/usr/bin/env node
// Example MCP client for Calliope TS — Streamable HTTP transport.
// Tests both local and remote HF Space endpoints.
//
// Usage:
//   node examples/mcp-client.mjs [url]
//   Default URL: http://localhost:7860/mcp
//   Remote: https://AlekseyCalvin-cts.hf.space/mcp
//
// Requires: npm install @modelcontextprotocol/sdk
// (already in dev deps of this repo via package.json)

import { Client } from '@modelcontextprotocol/sdk/client/index.js';
import { StreamableHTTPClientTransport } from '@modelcontextprotocol/sdk/client/streamableHttp.js';

const url = process.argv[2] || 'http://localhost:7860/mcp';
console.log(`Connecting to ${url} ...`);

const transport = new StreamableHTTPClientTransport(new URL(url));
const client = new Client({ name: 'calliope-mcp-example', version: '1.0.0' });

await client.connect(transport);
console.log('Connected. Server info:', await client.getServerVersion());

console.log('\n--- tools/list ---');
const { tools } = await client.listTools();
console.log(tools.map(t => `- ${t.name}: ${t.description.slice(0, 80)}...`).join('\n'));

console.log('\n--- scan_poem (summary) ---');
const poem = `Shall I compare thee to a summer's day?
Thou art more lovely and more temperate:
Rough winds do shake the darling buds of May,
And summer's lease hath all too short a date`;

let res = await client.callTool({
  name: 'scan_poem',
  arguments: { text: poem, detail_level: 'summary' }
});
console.log(res.content[0].text.slice(0, 3000));

console.log('\n--- find_rhymes (day) ---');
res = await client.callTool({
  name: 'find_rhymes',
  arguments: { word: 'day', limit: 10 }
});
console.log(res.content[0].text);

console.log('\n--- get_capabilities ---');
res = await client.callTool({
  name: 'get_capabilities',
  arguments: {}
});
console.log(res.content[0].text.slice(0, 1500));

await client.close();
console.log('\nDone.');

File: examples/mcp-client.py
#!/usr/bin/env python3
"""
Python MCP client for Calliope TS — Streamable HTTP
Requires: pip install mcp
Usage: python examples/mcp-client.py [url]
Default: http://localhost:7860/mcp
Remote: https://AlekseyCalvin-cts.hf.space/mcp
"""
import asyncio
import sys

try:
    from mcp import ClientSession
    from mcp.client.streamable_http import streamablehttp_client
except ImportError:
    print("Please install MCP Python SDK: pip install mcp")
    sys.exit(1)

URL = sys.argv[1] if len(sys.argv) > 1 else "http://localhost:7860/mcp"

async def main():
    print(f"Connecting to {URL} via Streamable HTTP...")
    async with streamablehttp_client(URL) as (read, write, _):
        async with ClientSession(read, write) as session:
            await session.initialize()
            print("Initialized")

            tools = await session.list_tools()
            print("\nTools:")
            for t in tools.tools:
                print(f" - {t.name}: {t.description[:80]}...")

            print("\n--- scan_poem ---")
            res = await session.call_tool("scan_poem", {
                "text": "Shall I compare thee to a summer's day?\nThou art more lovely and more temperate:",
                "detail_level": "summary"
            })
            print(res.content[0].text[:4000])

            print("\n--- parse_syntax ---")
            res = await session.call_tool("parse_syntax", {
                "text": "Because I could not stop for Death",
            })
            print(res.content[0].text[:3000])

            print("\n--- find_rhymes ---")
            res = await session.call_tool("find_rhymes", {"word": "time", "limit": 10})
            print(res.content[0].text)

if __name__ == "__main__":
    asyncio.run(main())
